java/code/src/com/redhat/rhn/domain/task/TaskFactory.java

```diff
@@ -129,6 +129,24 @@ public static void deleteByOrgNameDataPriority(Org org, String name, Long data,
         session.createQuery(criteriaDelete).executeUpdate();
     }
 
+    /**
+     * Delete tasks matching a name and a data.
+     *
+     * @param name the tasks name
+     * @param data the tasks data
+     */
+    public static void deleteByNameData(String name, Long data) {
```

Comment:
the code looks good to me. I would improve the Javadoc comment to mentione that it will ignore the organization, meaning it can delete cross-organization Jobs. And maybe mention it also in the method name?

Reply:
I'll let the name as it is: it fits the tacit naming convention in that file. I'll enhance the javadoc